Feature/update templates

catalinaIlabaca opened the pull request:
Se actualizaron las templates basadas en las del tema de eol-theme-2020, posterior a eso se eliminaron las referencias hacia el tema, actualizando una imagen y una referencia de una archivo.
Se modifica la manera de renderizar una template mako
Se cambian de lugar las templates
Se actualiza el setup.py para que obtenga las templates de static

catalinaIlabaca committed:
feat: update template from eol-uchile-2020

catalinaIlabaca committed:
refactor: mako templates

-  delete eol-theme-2020 references and add self reference
-  add mako template rendering
-  add include_mako_static to support variable substitution in templates

catalinaIlabaca committed:
test: update test  student_view_render to new mako_render format

catalinaIlabaca committed:
chore: update semantic version to 1.1.0

tomaspemora commented on eoldiscussion/static/html/_thread_list_template.html:
```diff
@@ -37,11 +37,11 @@
                 <span class="sr">${_("Sort:")}</span>
                 <select class="forum-nav-sort-control">
                     ## Translators: This is a menu option for sorting forum threads
-                    <option value="activity">${_("by recent activity")}</option>
```
No me quedan claros estos cambios de idioma. Vamos a dejarlo en español en duro? No estamos considerando traducciones desde el inglés en este bloque?

catalinaIlabaca replied:
Por ahora no, pues en el el tema estaban en español, como lo dice el commit iguale a lo que estaba en el tema, estoy de acuerdo con que deberiamos agregar las traducciones a este repo

tomaspemora replied:
hagamoslo en un siguiente PR.

tomaspemora reviewed (changes requested):
Creo que queda más ordenado el PR si el primer commit mueve las templates a la carpeta final en que quedarán, que en este caso sería el https://github.com/eol-uchile/eol_xblock_discussion/pull/2/changes/1581f86c1abff834edd5672114f464ad845abb7e y luego se aplican cambios sobre esas templates como en https://github.com/eol-uchile/eol_xblock_discussion/pull/2/changes/f02957473026ffbd1dbe82286da4f943d3cf68a9, https://github.com/eol-uchile/eol_xblock_discussion/pull/2/changes/fc3865e7517ed3943a40ec786aeafc4ed0798f41, etc. En esa misma linea, si uno mira el PR completo con los 9 commits, la mayoría de las templates se movieron pero en el caso de `_discussion_inline.html` esta se elimina y agrega. Para que quede consistente considerar usar `git mv`. Quizás esto se arregla moviendo el commit que hace el `git mv` primero.

catalinaIlabaca committed:
chore:  move templates and package to include statics

- move templates to static html directory -
- update package data to include static files

catalinaIlabaca commented:
@tomaspemora Listo lo solicitado, converse con vicente porque aun sale como si el archivo hubiera sido eliminado y creado, pero al revisar el blame se mantiene el historial, y vicente dijo que estaba hecho de la manera correcta

fsologureng commented:
@tomaspemora @catalinaIlabaca hasta donde yo entiendo, la convención es que la carpeta static es para templates que no son procesadas o para templates javascript (underscore), era imprescindible cambiarlas de lugar para que funcionaran?

tomaspemora commented:
> @tomaspemora @catalinaIlabaca hasta donde yo entiendo, la convención es que la carpeta static es para templates que no son procesadas o para templates javascript (underscore), era imprescindible cambiarlas de lugar para que funcionaran?

Creo que en el caso de los xblock es en static donde van las templates html procesadas también https://github.com/edx/cookiecutter-xblock/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html

https://github.com/openedx/edx-cookiecutters/blob/master/cookiecutter-xblock/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html

fsologureng commented:
> > @tomaspemora @catalinaIlabaca hasta donde yo entiendo, la convención es que la carpeta static es para templates que no son procesadas o para templates javascript (underscore), era imprescindible cambiarlas de lugar para que funcionaran?
> 
> Creo que en el caso de los xblock es en static donde van las templates html procesadas también https://github.com/edx/cookiecutter-xblock/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html
> 
> https://github.com/openedx/edx-cookiecutters/blob/master/cookiecutter-xblock/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html

Esos códigos son 2 años más antiguos que el 1er commit de este xblock, pero bueno, contesta la pregunta.

tomaspemora commented:
> > > @tomaspemora @catalinaIlabaca hasta donde yo entiendo, la convención es que la carpeta static es para templates que no son procesadas o para templates javascript (underscore), era imprescindible cambiarlas de lugar para que funcionaran?
> > 
> > 
> > Creo que en el caso de los xblock es en static donde van las templates html procesadas también https://github.com/edx/cookiecutter-xblock/blob/master/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html
> > https://github.com/openedx/edx-cookiecutters/blob/master/cookiecutter-xblock/%7B%7Bcookiecutter.repo_name%7D%7D/%7B%7Bcookiecutter.package_name%7D%7D/static/html/%7B%7Bcookiecutter.package_name%7D%7D.html
> 
> Esos códigos son 2 años más antiguos que el 1er commit de este xblock, pero bueno, contesta la pregunta.

Esa estructura no ha cambiado hasta donde yo sé
https://docs.openedx.org/projects/xblock/en/latest/xblock-tutorial/anatomy/html.html

fsologureng commented:
> Esa estructura no ha cambiado hasta donde yo sé https://docs.openedx.org/projects/xblock/en/latest/xblock-tutorial/anatomy/html.html

https://github.com/openedx/xblock-sdk/tree/master/sample_xblocks/thumbs/static
>  Each XBlock can choose its directory structure, as long as it specifies the correct paths to importlib.resources.

Pero claramente es la estructura de facto :+1:

tomaspemora commented on eoldiscussion/static/html/_underscore_templates2.html:
```diff
@@ -20,4 +20,4 @@
   % endif
 
 </script>
-<%static:include path="/openedx/themes/eol-uchile-2020/lms/templates/discussion/templates2.underscore" />
```
esta linea se modifica en:
- https://github.com/eol-uchile/eol_xblock_discussion/commit/5884963a342190de620b349028d0c84a55c2197e#diff-c34a2eff958840c765c2b5612831bcefb6b77b983e46609f9e4d3e7282af38b8L23
- https://github.com/eol-uchile/eol_xblock_discussion/commit/d30eb63976718f333a0300575c474095670ded52#diff-c34a2eff958840c765c2b5612831bcefb6b77b983e46609f9e4d3e7282af38b8L23
- https://github.com/eol-uchile/eol_xblock_discussion/commit/79a98e2a764263ad14fd3fd9fc10e84ecd05bab9#diff-c34a2eff958840c765c2b5612831bcefb6b77b983e46609f9e4d3e7282af38b8L23-R23
- https://github.com/eol-uchile/eol_xblock_discussion/pull/2/changes/652a60111ad7c20e8d61e8b737710e1f903f6072#diff-c34a2eff958840c765c2b5612831bcefb6b77b983e46609f9e4d3e7282af38b8L23

Creo que es innecesario tener tantas modificaciones intermedias de ese archivo al menos en esa línea. Si hay una lógica como:
1. Cambio las templates de lugar
2. Reemplazo las templates actuales de este repo por las más actualizadas que vivían en el tema
3. Actualizó las templates al valor final con el que debe quedar en este repo

Me parece bien, pero me da la impresión de que hay modificaciones que están demás.

Por favor revisar si no hay otro caso similar.

tomaspemora reviewed (changes requested):
En mi opinión los commits:
- https://github.com/eol-uchile/eol_xblock_discussion/commit/652a60111ad7c20e8d61e8b737710e1f903f6072
- https://github.com/eol-uchile/eol_xblock_discussion/commit/79a98e2a764263ad14fd3fd9fc10e84ecd05bab9
- https://github.com/eol-uchile/eol_xblock_discussion/commit/d42a7ae20aba536294d49b53566965cb7879418d

Podrían squashearse bajo la consigna "mako templates"

tomaspemora commented on setup.py:
```diff
@@ -37,5 +37,5 @@ def package_data(pkg, roots):
             'eoldiscussion = eoldiscussion:EolDiscussionXBlock',
         ],
     },
-    package_data={"eoldiscussion": ["*.html","*.underscore"]}
```
esto lo movería con el primer commit

tomaspemora commented:
diría que estamos ok para el merge @catalinaIlabaca confirmar por favor

catalinaIlabaca commented:
@tomaspemora confirmo correcto el squash
